File: C Code/Jigar_sir/C++_Practical/Hybrid_Inheritance.cpp
#include<iostream>
#include<conio.h>
using namespace std;
class A
{int a;
	public:
		void getA()
		{
			cout<<"\nEnter A: ";
			cin>>a;
		}
		void putA()
		{
			cout<<"\n A : "<<a;
		}
	
};
class B:public A
{	int b;
	public:
		void getB()
		{
			cout<<"\nEnter B: ";
			cin>>b;
		}
		void putB()
		{
			cout<<"\n B : "<<b;
		}
};
class c:public A
{	int c;
	public:
		void getC()
		{
			cout<<"\nEnter c: ";
			cin>>c;
		}
		void putC()
		{
			cout<<"\n C : "<<c;
		}
};
class D:public B,public c
{	int d;
	public:
		void getD()
		{
			cout<<"\nEnter d: ";
			cin>>d;
		}
		void putD()
		{
			cout<<"\n D : "<<d;
		}
};
int main()
{
	D d1;
	//d1.getA(); can't call
	d1.getB();
	d1.getC();
	d1.getD();
	//d1.putA(); can't call
	d1.putB();
	d1.putC();
	d1.putD();
	getch();
}

File: Python/Practice/HybridInheritance.py
class A:
    def getA(self,a):
        self.a=a
    def putA(self):
        print("A :", self.a)
class B(A):
    def getB(self,b):
        self.b=b
    def putB(self):
        print("B :",self.b)

class C(A):
    def getC(self,c):
        self.c=c
    def putC(self):
        print("C :",self.c)

class D(B,C):
    def getD(self,d):
        self.d=d
    def putD(self):
        print("D :",self.d)
        
b1=B()
b1.getA(10)
b1.putA()
b1.getB(20)
b1.putB()
c1=C()
c1.getA(10)
c1.putA()
c1.getC(30)
c1.putC()
d1=D()
d1.getA(10)
d1.putA()
d1.getB(20)
d1.putB()
d1.getC(30)
d1.putC()
d1.getD(40)
d1.putD()


